internal/pkg/cli/pipeline_init.go

```diff
@@ -239,31 +246,65 @@ func (o *initPipelineOpts) askEnvs() error {
 		}
 		o.environments = envs
 	}
+
+	var envConfigs []*config.Environment
+	for _, environment := range o.environments {
+		envConfig, err := o.store.GetEnvironment(o.appName, environment)
+		if err != nil {
+			return fmt.Errorf("get config of environment: %w", err)
+		}
+		envConfigs = append(envConfigs, envConfig)
+	}
+	o.envConfigs = envConfigs
+
 	return nil
 }
 
 func (o *initPipelineOpts) askRepository() error {
 	var err error
 	if o.repoURL == "" {
-		if err = o.selectGitHubURL(); err != nil {
+		if err = o.selectURL(); err != nil {
 			return err
 		}
 	}
-	if o.githubOwner, o.githubRepo, err = o.parseOwnerRepoName(o.repoURL); err != nil {
-		return err
-	}
-	if o.githubAccessToken == "" {
-		if err = o.getGitHubAccessToken(); err != nil {
+
```

Comment:
nit: what do you think of refactoring like this?
```go
switch {
case strings.Contains(o.repoURL, githubURL):
    return o.askGitHubRepoDetails()
case strings.Contains(o.repoURL, codecommitURL):
    return o.askCodeCommitRepoDetails()
}
```

Reply:
Done!